jira-test/jira-test-docker/src/test/it/org/xwiki/contrib/jira/test/ui/PastePluginIT.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.jira.test.ui;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.xwiki.administration.test.po.AdministrationPage;
+import org.xwiki.ckeditor.test.po.CKEditor;
+import org.xwiki.contrib.jira.test.po.JIRAAdministrationSectionPage;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.test.docker.junit5.UITest;
+import org.xwiki.test.ui.TestUtils;
+import org.xwiki.test.ui.po.ViewPage;
+import org.xwiki.test.ui.po.editor.WYSIWYGEditPage;
+import org.xwiki.test.ui.po.editor.WikiEditPage;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Verify the functionality of the CKEditor paste plugin.
+ *
+ * @version $Id$
+ * @since 10.0
+ */
+@UITest
+public class PastePluginIT
+{
+    private static final String ADMIN_CONFIG_JIRA_ID = "testId";
+
+    private static final String ADMIN_CONFIG_JIRA_URL = "https://localhost:8889";
```

Comment:
How does that work since there's no local jira installed?

If you check `JIRAMacroIT` you'll see that I've solved this with WireMock.

Reply:
Yes. I initially started creating the tests with that in mind but it occured to me that the plugin is responsible with the correct mapping between a jira link and a jira macro with adequate parameters set. Whether the jira macro displays correctly or not is not in the scope of this functionality and is covered by `JiraMacroIT`. This is why I dropped the WireMock setup. If you think that it's important for this test to also cover the correct displaying of the macro, I can go ahead and cover that as well.

Reply:
My question was more: what is this URL? And how can it work? :)

EDIT: After reading the test more I think it's just a placeholder and it's not used. I got tricked by the value. Why not instead: `https://jirainstancedomain`?

Re testing the display, I've now checked your test. It looks fine to me, even if there's still possibly a path that is not tested (ie that the paste will lead to the jira macro executing in the display immediately, without any refresh), i.e. an end to end test basically. I think it would be nicer and since WireMock is local it shouldn't impact the speed of the test much at all. WDYT?

Thx!

Reply:
It would be nicer indeed. I'm looking into it.